### PR TITLE
Add Min/Max Window Size Setting

### LINKED ISSUE #31303: Minimal and maximal window size in project settings
**Godot version:** 3.1.1

**Issue description:**
Hey guys,
so I was wondering why there isn't an option for the minimal and the maximal window size in the  project settings.

I mean it's possible to solve this with a gdscript but I believe it would be a nice feature.

Regards
BaseCrusher

### PR DESCRIPTION
**Pull request by codecustard**
Add project settings to easily set the minimum and maximum window size.

Closes: #31303

Related PR that caused crashing: #31367

### CODE REVIEW

File: main/main.cpp [hunk 1]
```diff
@@ -1152,6 +1160,18 @@ Error Main::setup2(Thread::ID p_main_tid_override) {
 		OS::get_singleton()->set_window_always_on_top(true);
 	}
 
+	int mw = GLOBAL_GET("display/window/size/min_width");
+	int mh = GLOBAL_GET("display/window/size/min_height");
+	OS::get_singleton()->set_min_window_size(Size2(mw, mh));
+
+	mw = GLOBAL_GET("display/window/size/max_width");
+	if (mw <= 0)
```
Comment (akien-mga) [inline main/main.cpp]: If 0, there should not be a max window size IMO.

**Comment by aaronfranke** [comment]
@codecustard  Is this still desired? If so, it needs to be rebased on the latest master branch. Also, the above feedback should be addressed.

Otherwise, abandoned pull requests will be closed in the future as announced [here](https://godotengine.org/article/headsup-vulkan-merged-master-branch-unstable).

**Comment by follower** [comment]
Certainly the minimum window size aspect of this is still desired.

Given the implementation question around the maximum size, I think it would make sense to split the maximum-window-size related functionality into a separate PR so at least the minimum window size can be enforced.

Being able to shrink the window in such a way that UI controls are cut-off looks...unpolished IMO. :)

**Comment by aaronfranke** [comment]
This PR is abandoned, closing. Anyone is welcome to salvage this, however, note that there is overlap with #37260.

**Pull request closed without merging**